### ISSUE TITLE
Inclusion of ‘default’ entries (eg 01/01/2025 for date boxes)

### ISSUE THREAD

**Issue by IanMayo** [serge-web/serge#916]
Discussed with Mark on 30/6/21

The intention for this was to overcome the issue of players mistkenly entering dates that were several days/months away from the intended time.

The solution is to provide the date-time of the start of the current turn as the default value for any date-time field.

For some field representing a time that doesn't occur within the current turn, the control will still have the rough date-time, which should still help with accuracy.

To implement this for messages from templates, we _may_ have to inspect/mangle the form template, to look for a text field formatted as date-time, then to `inject` the current date-time (`gameTurnTime`) as a default value.

**Comment by sanjayradadiya** [comment]
Do we need to update this thing inside JSON editor like add current DateTime ?

**Comment by IanMayo** [comment]
Aah, hold on.  I was mistaken.  I did think we had to modify the DOM elements returned from `jsoneditor`.  That's wrong. We have to modify the template (schema) that is provided to `jsoneditor` - to provide a `default` value if the format is `datetime`

I _guess_ our code will:
* clone the template
* walk the JSON object structure, looking for any elements with "format": "datetime" or "format": "date"
* insert a "default" value of the current `gameTurnTime` or just the date portion of `gameTurnTime`.
* pass the new template to the `jsoneditor` constructor.
